Final win condition

drdreo opened the pull request:
Implemented the victory points win condition for headquarters.
Added Defeat confetti.
Normalized lobbyKey on server.
Player Stats are now shown in the Dashboard

drdreo committed:
player stats implemented in sidebar
added final Victory Condition (VP)

drdreo committed:
normalized lobbyKey
safeguard for attacking or Expanding

drdreo committed:
implemented Defeat Confetti Ash

drdreo committed:
fixed schema error in unit test

davidHolzmann commented on server/models/Game.ts:
```diff
@@ -615,7 +601,8 @@ export class Game implements IGame {
   occupyCountry(countryId: number, occupier: Player): void {
     this._map.countries[countryId].occupier = occupier.id;
-    occupier.occupyCountry(countryId);
+    const hasHQ = !!this.map.countries[countryId].headQuarter;
```
refactor and take data from inside the Player

davidHolzmann reviewed (changes requested):
change described in comment

drdreo committed:
fixed sidebar menu
fixed battle view responsiveness
changed request review to grab data inside player fixes https://github.com/drdreo/risikonline/pull/189#discussion_r304343647

drdreo committed:
Merge branch 'master' into finalWinCondition

davidHolzmann commented on client/src/app/game/game.component.html:
```diff
@@ -36,5 +36,7 @@
-<end-screen *ngIf="gameService.gameEnded"
-            [winner]="gameService.winner.username === playerService.player.username"></end-screen>
+
+<end-screen *ngIf="gameService.gameEnded" [winner]="gameService.winner.username"></end-screen>
```
winner has type boolean!

drdreo committed:
fixed the merge fail
